synapse/handlers/federation_event.py

```diff
@@ -1773,7 +1733,7 @@ async def _maybe_kick_guest_users(self, event: EventBase) -> None:
     async def _check_for_soft_fail(
         self,
         event: EventBase,
-        state_ids: Optional[StateMap[str]],
+        context: EventContext,
         origin: str,
     ) -> None:
```

Comment:
It only makes sense to call `_check_for_soft_fail` prior to persisting an event and `EventContext`s are intended to hold information relevant to persisting an event. So we aren't losing much by accepting an `EventContext` instead of a `StateMap`.